cli/cmd/sync.go

```diff
@@ -58,49 +57,47 @@ func sync(cmd *cobra.Command, args []string) error {
 			}
 			destinationsSpecs = append(destinationsSpecs, *spec)
 		}
-		if err := syncConnection(ctx, pm, *sourceSpec, destinationsSpecs); err != nil {
+		if err := syncConnection(ctx, *sourceSpec, destinationsSpecs); err != nil {
 			return fmt.Errorf("failed to sync source %s: %w", sourceSpec.Name, err)
 		}
 	}
 
 	return nil
 }
 
-func syncConnection(ctx context.Context, pm *plugins.PluginManager, sourceSpec specs.Source, destinationsSpecs []specs.Destination) error {
-	sourcePlugin, err := pm.NewSourcePlugin(ctx, sourceSpec.Registry, sourceSpec.Path, sourceSpec.Version)
+func syncConnection(ctx context.Context, sourceSpec specs.Source, destinationsSpecs []specs.Destination) error {
+	sourceClient, err := clients.NewSourceClient(ctx, sourceSpec.Registry, sourceSpec.Path, sourceSpec.Version)
 	if err != nil {
 		return fmt.Errorf("failed to get source plugin client for %s: %w", sourceSpec.Name, err)
 	}
-	defer sourcePlugin.Close()
-	sourceClient := sourcePlugin.GetClient()
+	defer sourceClient.Close()
 
-	destPlugins := make([]*plugins.DestinationPlugin, len(sourceSpec.Destinations))
+	destClients := make([]*clients.DestinationClient, len(sourceSpec.Destinations))
 	destSubscriptions := make([]chan []byte, len(sourceSpec.Destinations))
 	for i := range destSubscriptions {
 		destSubscriptions[i] = make(chan []byte)
 	}
 	defer func() {
-		for _, destPlugin := range destPlugins {
-			if destPlugin != nil {
-				destPlugin.Close()
+		for _, destClient := range destClients {
+			if destClient != nil {
+				destClient.Close()
```

Comment:
Not directly related to the work in this PR, but technically this `Close` returns an error that we're ignoring, we may want to at least log it

Reply:
Good catch. Print is good - log is already done on the sdk side (though I should add some more there)